packages/expo-notifications/ios/EXNotifications/Notifications/Categories/CategoriesModule.swift

```diff
@@ -4,45 +4,53 @@ import ExpoModulesCore
-      UNUserNotificationCenter.current().getNotificationCategories { categories in
-        let existingCategories = categories.map { category in
-          return CategoryRecord(category)
-        }
-        promise.resolve(existingCategories)
-      }
+      let categories = await UNUserNotificationCenter.current().notificationCategories()
+      promise.resolve(filterAndSerializeCategories(Array(categories)))
     }
 
     AsyncFunction("setNotificationCategoryAsync") { (identifier: String, actions: [CategoryActionRecord], options: CategoryOptionsRecord?, promise: Promise) in
-      let categoryRecord = CategoryRecord(identifier, actions: actions, options: options)
-      let newNotificationCategory = categoryRecord.toUNNotificationCategory()
-      UNUserNotificationCenter.current().getNotificationCategories { oldcategories in
-        let newCategories = Set(oldcategories.filter { oldCategory in
-          return oldCategory.identifier != newNotificationCategory.identifier
-        }
-        .union([newNotificationCategory]))
-        UNUserNotificationCenter.current().setNotificationCategories(newCategories)
-        promise.resolve(CategoryRecord(newNotificationCategory))
-      }
+      setNotificationCategoryAsync(identifier: identifier, actions: actions, options: options, promise: promise)
     }
 
     AsyncFunction("deleteNotificationCategoryAsync") { (identifier: String, promise: Promise) in
-      UNUserNotificationCenter.current().getNotificationCategories { oldCategories in
-        let didDelete = oldCategories.contains { oldCategory in
-          return oldCategory.identifier == identifier
-        }
-        if didDelete {
-          let newCategories = Set(oldCategories.filter { oldCategory in
-            return oldCategory.identifier != identifier
-          })
-          UNUserNotificationCenter.current().setNotificationCategories(newCategories)
-        }
-        promise.resolve(didDelete)
+      deleteNotificationCategoryAsync(identifier: identifier, promise: promise)
+    }
+  }
+
+  open func filterAndSerializeCategories(_ categories: [UNNotificationCategory]) -> [CategoryRecord] {
+    return categories.map { CategoryRecord($0) }
+  }
+
+  open func setNotificationCategoryAsync(identifier: String, actions: [CategoryActionRecord], options: CategoryOptionsRecord?, promise: Promise) {
+    let categoryRecord = CategoryRecord(identifier, actions: actions, options: options)
+    let newNotificationCategory = categoryRecord.toUNNotificationCategory()
+    UNUserNotificationCenter.current().getNotificationCategories { oldcategories in
+      let newCategories = Set(oldcategories.filter { oldCategory in
+        return oldCategory.identifier != newNotificationCategory.identifier
+      }
+      .union([newNotificationCategory]))
+      UNUserNotificationCenter.current().setNotificationCategories(newCategories)
+      promise.resolve(CategoryRecord(newNotificationCategory))
+    }
+  }
```

Comment:
```suggestion
  
  func setNotificationCategoryAsync(identifier: String, actions: [CategoryActionRecord], options: CategoryOptionsRecord?) async -> CategoryRecord {
    let categoryRecord = CategoryRecord(identifier, actions: actions, options: options)
    let newNotificationCategory = categoryRecord.toUNNotificationCategory()
    let oldcategories = await UNUserNotificationCenter.current().notificationCategories()
    let newCategories = Set(oldcategories.filter { oldCategory in
      return oldCategory.identifier != newNotificationCategory.identifier
    }.union([newNotificationCategory]))
    UNUserNotificationCenter.current().setNotificationCategories(newCategories)
    return CategoryRecord(newNotificationCategory)
  }
```
Let's make this async and return the result